src/main/java/net/kodehawa/mantarobot/commands/MoneyCmds.java

```diff
@@ -108,151 +110,159 @@ public void call(GuildMessageReceivedEvent event, I18nContext languageContext, S
-                            streak = languageContext.withRoot("commands", "daily.streak.first_time");
-                        } else {
-                            streak = String.format(languageContext.withRoot("commands", "daily.streak.lost_streak"), authorPlayerData.getDailyStreak());
-                        }
 
-                        authorPlayerData.setDailyStreak(1);
-                    }
+                    toAddMoneyTo = UnifiedPlayer.of(otherUser, getConfig().getCurrentSeason());
 
-                    if (authorPlayerData.getDailyStreak() > 5) {
-                        int bonus = 150;
 
-                        if (authorPlayerData.getDailyStreak() > 15)
-                            bonus += Math.min(1700, Math.floor(150 * authorPlayerData.getDailyStreak() / 10D));
+                } else{
+                    // This is here so you dont overwrite yourself....
+                    authorPlayer = toAddMoneyTo.getPlayer();
+                    authorPlayerData = authorPlayer.getData();
+                }
 
-                        streak += String.format(languageContext.withRoot("commands", "daily.streak.given.bonus"), (mentionedUser == null ? "You" : mentionedUser.getName()), bonus);
-                        money += bonus;
-                    }
+                List<String> returnMessage = new ArrayList<String>();
+                long currentTime = System.currentTimeMillis();
+                int amountStreaksavers = authorPlayer.getInventory().getAmount(Items.MAGIC_WATCH);
+                // >=0 -> Valid  <0 -> Invalid
+                long currentDailyOffset = DAILY_VALID_PERIOD_MILLIS - (currentTime - authorPlayerData.getLastDailyAt()) ;
 
-                    if (authorPlayerData.getDailyStreak() > 10) {
-                        authorPlayerData.addBadgeIfAbsent(Badge.CLAIMER);
-                    }
+                long streak = authorPlayerData.getDailyStreak();
 
-                    if (authorPlayerData.getDailyStreak() > 100) {
-                        authorPlayerData.addBadgeIfAbsent(Badge.BIG_CLAIMER);
+                // Not expired?
+                if(currentDailyOffset + amountStreaksavers*DAILY_VALID_PERIOD_MILLIS >= 0 ){ ;
+                    streak++;
+                    returnMessage.add(String.format(languageContext.withRoot("commands", targetOther ? "daily.streak.given.up" : "daily.streak.up" ), streak));
+                    if(currentDailyOffset<0){
+                        int streaksaversUsed = 1;
+                        for(; currentDailyOffset + streaksaversUsed*DAILY_VALID_PERIOD_MILLIS < 0 && streaksaversUsed<=streaksaversUsed;streaksaversUsed++);
+                        authorPlayer.getInventory().process(new ItemStack(Items.MAGIC_WATCH,streaksaversUsed * -1));
+                        returnMessage.add(String.format(languageContext.withRoot("commands", "daily.streak.watch_used"), streaksaversUsed, streaksaversUsed+1, amountStreaksavers-streaksaversUsed));
```

Comment:
spacing again, should look like `(new ItemStack(Items.MAGIC_WATCH, streakSaversUsed * -1)`